include/git2/index.h

```diff
@@ -457,6 +457,38 @@ GIT_EXTERN(int) git_index_entry_stage(const git_index_entry *entry);
 GIT_EXTERN(int) git_index_add_bypath(git_index *index, const char *path);
 
 /**
+ * Add or update an index entry from a buffer in memory
+ *
+ * This method will create a blob in the repository that owns the
+ * index and then add the blob to the index.  The `path` of the entry
+ * represents the position of the blob relative to the repository's
+ * root folder.
```

Comment:
This wording seems off. You don't add a blob to the index, you add an entry. The blob is also pure content. It does not have a path at all.